### PR TITLE
Change the process by which the LastUsedID block works

### PR DESCRIPTION
**Pull request by StormiFire**
### What's changed
- This PR changes the way the `LastUsedID` block works by removing the `LAST_ID` variable and using a more native method.

### CODE REVIEW

**Comment by ysfchn** [comment]
This will also change the block behavior, because when a component has removed, LastUsedID will return another ID because instead of returning **last used** ID, it returns **last component's** ID.

**Comment by StormiFire** [comment]
> This will also change the block behavior, because when a component has removed, LastUsedID will return another ID because instead of returning **last used** ID, it returns **last component's** ID.

You cannot consider something last, if it's been removed. It's expected that creating a new component will update the last id, so that others don't have to.

**Comment by ysfchn** [comment]
> You cannot consider something last, if it's been removed. 

I'm just saying it will change the block's behavior which users may need to modify their existing projects.

**Comment by StormiFire** [comment]
> > You cannot consider something last, if it's been removed.
> 
> I'm just saying it will change the block's behavior which users may need to modify their existing projects.

Yes, but this won't majorly break projects... A block won't disappear.

File: src/com/yusufcihan/DynamicComponents/DynamicComponents.java [hunk 1]
```diff
@@ -444,7 +433,10 @@ public void Remove(String id) {
     */
     @SimpleFunction(description = "Returns the last created component's ID by Create block.")
     public String LastUsedID() {
-        return LAST_ID;
+        List COMPONENT_IDS = Arrays.asList(COMPONENTS.keySet().toArray());
+        int size = COMPONENT_IDS.size();
+
+        return (size > 0 ? COMPONENT_IDS.get(size - 1).toString() : "");
```
Comment (ysfchn) [inline src/com/yusufcihan/DynamicComponents/DynamicComponents.java:439]: How about shortening it?

```suggestion
        Object[] COMPONENT_IDS = COMPONENTS.keySet().toArray();
        return (COMPONENT_IDS.length > 0 ? COMPONENT_IDS[COMPONENT_IDS.length - 1].toString() : "");
```

And by the way we already have `java.util.Arrays;` import in the Line 25.

Comment (StormiFire) [reply src/com/yusufcihan/DynamicComponents/DynamicComponents.java:439]: Hello, sorry for the last response. I've been very busy these past few days working on other things off the Kodular platform and out of my own comfort zone. I should be free this weekend, so if you can hold off on **2.1.0** until Saturday or Sunday, that would be appreciated.

File: src/com/yusufcihan/DynamicComponents/DynamicComponents.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (ysfchn) [inline src/com/yusufcihan/DynamicComponents/DynamicComponents.java:20]:
```suggestion
```

Can you remove the import there as it is not needed anymore?

**Comment by ysfchn** [comment]
The last thing, it causes a conflict right now. When you fetch the latest commit, it will become ready for merge.

**Comment by StormiFire** [comment]
Resolved conflict, ready for review.

**Comment by ysfchn** [comment]
Thanks!

**Pull request merged**